core/jobs/batch_jobs/exp_migration_jobs.py

```diff
@@ -509,3 +510,177 @@ def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
             )
             | beam.Flatten()
         )
+
+
+# TODO(#15927): This job needs to be kept in sync with MigrateExplorationJob and
+# later we will unify these jobs together.
+class AuditExplorationMigrationJob(base_jobs.JobBase):
```

Comment:
@vojtechjelinek Just a thought: Instead of writing a separate Audit job, should we just avoid the put call (line 492) in case of any validation error?

Benefits:
- Avoid duplicate code 
- Avoid extra steps to run the audit job beforehand

Reply:
Yes, this is just a temporary code that will later be deduplicated.

Reply:
Hi @vojtechjelinek -- I think Sandeep is asking something different here.

What you are saying in that issue is that you want to have common functions for both jobs in order to reduce duplication.

What @DubeySandeep is asking (and I think @kevintab95 agrees) is to have just a single job. That job does the full audit and then in its next step it actually does the puts. If any exp fails then no migrations are carried out at all and the errors are shown to the release coordinator.

Is that possible? That will keep things safe and seems like it would reduce the chance of errors.

Reply:
Yeah, that might be possible. I would still say lets first merge this and do that later though.